src/components/flashcard-set/flashcard-set.tsx

```diff
@@ -30,16 +34,16 @@ export const FlashcardSet = ({
   function getCards() {
     return cards.map((card, index) => {
       return (
-        <Reorder.Item key={card.id ?? card.key} value={card}>
+        <Reorder.Item key={card.key} value={card}>
```

Comment:
this is assuming that when we pull cards from the API, we set key = id?

in other words, the invariant:
- if new card on front end:
  - id: undefined 
  - key: uuidv4()
- if card fetched from API:
  - id = key

Reply:
I was assuming that they would always remain separate. So every card would have a `key` that is a constant uuid that we create after fetching the cards.

When we add a card to the database with a fetch call, we would then only update the id on that card.

Reply:
ah I see, if we're okay with the DOM being completely reset when `Save` is clicked, let's proceed
especially not a concern if we're going the route with **not** refetching on save as you mentioned in Discord

Reply:
Wait, it would be completely reset? I thought if we didn't re-fetch the cards, only the newly added cards would be re-rendered, which would be the same either way, right?

Reply:
oh yeah, that's what i meant. not a concern if we're not refetching anymore